09-schema/03-rules.md

```diff
@@ -274,6 +274,112 @@ TypeQLDefine query = TypeQL.define(
 </div>
 
 This rule will make every relation transitive.
+## Efficient transitivity
+
+A common use-case for rules is to infer the transitive closure of a relation. The most straight-forward way of doing this is as follows:
+```typeql
+define
+
+rule transitive-reachability:
+when{
+    (from: $x, to: $y) isa edge;
+    (from: $y, to: $z) isa edge;
+} then {
+    (from: $x, to: $z) isa edge;
+};
+```
+We can interpret this rule as joining two paths together. In a chain `p-q-r-s-t`, to find all nodes reachable from p, we would generate the following relations:
+```
+p--q, q--r, r--s, s--t  (already in the database)
+
+p--r, q--s, r--t,       (Inferred)
+p--s, q--t, p--t        (Inferred)                    
+```
+
+Concretely, We would generate an `edge` relation _for every pair_ of nodes reachable from `p` - a **quadratic** number of relations.
+
+ The following section describes a recipe to answer forward transitivity queries materialising only a **linear** number of relations. Later, the recipe is extended to backward queries and undirected relations.
+
+### Forward transitivity
+We first define separate types for the persisted and (inferred) transitive version of the relation.
```

Comment:
show how as a schema snippet. To facilitate this, I think you include a the type definition snippet with the naiive rule you show above